exporter/awsprometheusremotewriteexporter/auth.go

```diff
@@ -147,3 +158,20 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+// addToUserAgent will format the provided name and version to the name/version format.
+// If the extra parameters are provided they will be added as metadata to the
+// name/version pair resulting in the following format.
+// "name/version (extra0; extra1; ...)"
+// It then adds the string to the end of the request's current User-Agent.
+func addToUserAgent(req *http.Request, name, version string, extra ...string) {
+	ua := fmt.Sprintf("%s/%s", name, version)
```

Comment:
Use strings.Builder to reduce allocations

https://golang.org/pkg/strings/#Builder

Reply:
Actually looks like we can compute a string as a constant containing the sdk / runtime stuff. Then just the last five lines need to happen when signing.

Reply:
Yes, I will compute the string as a constant and store it in the `signingRoundTripper` struct, and from there append it to the User-Agent header when signing.